sdk/v2/core/projects.go

```diff
@@ -340,3 +341,15 @@ func (p *projectsClient) Authz() AuthzClient {
 func (p *projectsClient) Secrets() SecretsClient {
 	return p.secretsClient
 }
+
+// ValidateProjectID checks if a given id is valid.
+func ValidateProjectID(id string) error {
+	regex := regexp.MustCompile(`^[a-z][a-z\d-]*[a-z\d]$`)
```

Comment:
This regex will get recompiled every time this function is called. It would be better to move this to the package level where it will be compiled once when the package is initialized. That was there's less overhead if and when this function is called repeatedly.